### PR TITLE
Entity game state

### PR DESCRIPTION
**Pull request by JJendryka**

### CODE REVIEW

File: src/entity/Entity.cpp [hunk 1]
```diff
@@ -39,16 +39,26 @@ Entity::Entity(nlohmann::json json) {
...
         sprite.setTexture(texture);
     }
-
-
-
 }
 
-Entity::Entity(int ID, std::string name, sf::Vector2f position,sf::RectangleShape hitbox, sf::Sprite sprite) {
+Entity::Entity(int ID, std::string name, sf::Vector2f position, sf::FloatRect hitbox, sf::Texture texture) {
    this->ID = ID;
```
Comment (Bravo555) [inline src/entity/Entity.cpp]: could we make this lowercase possibly?

File: src/entity/Entity.cpp [hunk 2]
```diff
@@ -39,16 +39,26 @@ Entity::Entity(nlohmann::json json) {
...
    this->name = std::move(name);
-   this->sprite = std::move(sprite);
-   this->hitbox = std::move(hitbox);
+   this->texture = texture;
+   this->sprite = sf::Sprite(texture);
+   this->hitbox = hitbox;
    this->position = position;
 }
 
+void Entity::render(sf::RenderWindow *window) {
```
Comment (Bravo555) [inline src/entity/Entity.cpp]: I thought we weren't supposed to use plain C++ pointers. Wouldn't a reference or `std::shared_pointer` be better in this case?

File: src/entity/Player.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    if (!touching_ground) {
+        speed.y -= 0.005f * deltaTime.count() / MOVEMENT_DIV;
+    }
+
+    position.y += deltaTime.count() * speed.y / MOVEMENT_DIV;
+
+
+    if (TileDatabase::get()[gameState->world.getTile((int)position.x / Chunk::TILE_SIZE, div((int)position.y, Chunk::TILE_SIZE)).tileId].isSolid) {
```
Comment (Bravo555) [inline src/entity/Player.cpp]: Okay, from what I see in this line we check if current tile is solid, but this line is long, hard to understand at first glance.

We should preferably make it look something like this:
```cpp
auto currentTile = world.getTile((int)position.x / Chunk::TILE_SIZE, div((int)position.y, Chunk::TILE_SIZE));

if(currentTile.isSolid()) {
// ...
}
```

Comment (Bravo555) [inline src/entity/Player.cpp]:
```cpp
world.getTile((int)position.x / Chunk::TILE_SIZE, div((int)position.y, Chunk::TILE_SIZE))
```
Why do you simply divide `x` coord but with `y` you use `div()` function?

File: src/entity/Player.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    speed = {0.0f, 0.0f};
+}
+
+void Player::update(std::chrono::microseconds deltaTime) {
+
+    auto div = [](int a, int b)->int{int ret = a/b; return ret>=0? ret: ret-1;};
+
+    const float MOVEMENT_DIV = 2000.0f;
+
+    if(sf::Keyboard::isKeyPressed(sf::Keyboard::W) && touching_ground){
```
Comment (Bravo555) [inline src/entity/Player.cpp]: We should probably use State Machine of some sort to manage state of entities.

Comment (Bravo555) [reply src/entity/Player.cpp]: Right now, one flag like that is fine, but we need to remember to take care of that in the near future.

Comment (JJendryka) [reply src/entity/Player.cpp]: Player won't have another state. There will be a lot more states in NPC. There we will make state machine.

File: src/entity/Player.cpp (hunk 3 above)
Comment (JJendryka) [reply src/entity/Player.cpp]: There is a huge mess in our coordinate system. I'm waiting for zakrent to finish and merge biomes to start fixing it.

**Pull request merged**